source/val/validate_conversion.cpp

```diff
@@ -273,6 +273,18 @@ spv_result_t ConversionPass(ValidationState_t& _, const Instruction* inst) {
           return _.diag(SPV_ERROR_INVALID_DATA, inst)
                  << "Pointer storage class must be PhysicalStorageBufferEXT: "
                  << spvOpcodeString(opcode);
+
+        if (spvIsVulkanEnv(_.context()->target_env)) {
+          if (!_.IsIntScalarType(input_data_type) ||
+              (_.GetBitWidth(input_data_type) != 64)) {
+            return _.diag(SPV_ERROR_INVALID_DATA, inst)
+                   << _.VkErrorID(4710)
+                   << "Pointer must be a 64 bit width integer with "
```

Comment:
This message seems confusing to me. How about:

> PhysicalStorageBuffer64 addressing requires the input integer have a 64-bit width.

Also, drop the opcode. The validator prints the instruction in question so the opcode is superfluous. The rest of the checks in this file originate from before that time.